include/onnxruntime/core/providers/dml/dml_provider_factory.h

```diff
@@ -30,6 +30,23 @@
 extern "C" {
 #endif
 
+enum class OrtDmlPerformancePreference {
+  Default,
+  HighPerformance,  // Default
+  LowPower
+};
+
+enum class OrtDmlDeviceFilter {
+  None,
+  Gpu,  // Default
+  Npu
+};
+
+struct OrtDmlDeviceOptions {
+  OrtDmlPerformancePreference p;
```

Comment:
>; [](http://example.com/codeflow?start=31&length=1)

use better names than p and f

Reply:
> use better names than p and f

👍👍 Except even further than than, follow the rule of API naming - use whole words, not word fragments.

```
  OrtDmlPerformancePreference performance_pref;
  OrtDmlDeviceFilter device_filter;
```

It's also common practice to name a field the same as its type (minus the C name scoping prefix of course).

Reply:
Similarly, `device_options` below. With IDE autocomplete, we don't really save any typing by using cryptic names.

(and yes, it should have been `ID3D12CommandQueue* command_queue` too)